moveit_core/collision_detection_bullet/src/collision_env_bullet.cpp

```diff
@@ -187,8 +187,29 @@ void CollisionEnvBullet::checkRobotCollisionHelper(const CollisionRequest& req,
         cow->getName(), state.getAttachedBody(cow->getName())->getGlobalCollisionBodyTransforms()[0]);
   }
 
+  // Specify active collision links
+  if (!req.group_name.empty())
+  {
+    auto joint_model_group = robot_model_->getJointModelGroup(req.group_name);
+    std::vector<std::string> active_links = joint_model_group->getLinkModelNames();
+    // Disable links that are not in the joint_model_group
+    for (std::string link_name : active_)
+    {
+      auto pos = std::find(active_links.begin(), active_links.end(), link_name);
+      if (pos == active_links.end())
+      {
+        manager_->disableCollisionObject(link_name);
+      }
+    }
+  }
+
   manager_->contactTest(res, req, acm, false);
 
+  // Reset
+  if (!req.group_name.empty())
+  {
+    ;//manager_->setActiveCollisionObjects(active_);  // All links
+  }
```

Comment:
One thing I'm struggling with is resetting after an iteration. I'd like to reset the active links like this, but doing so seems to null the joint_model_group specification on the next iteration --

```
for(auto link_name : active_)
      manager_->enableCollisionObject(link_name);
```

Reply:
ah, fixed it in 240a9630d